src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AuditMessage.java

```diff
@@ -307,15 +307,11 @@ public void addRestParams(Map<String,String> params) {
 
     public void addRestHeaders(Map<String,List<String>> headers, boolean excludeSensitiveHeaders) {
         if(headers != null && !headers.isEmpty()) {
-            if(excludeSensitiveHeaders) {
-                final Map<String, List<String>> headersClone = new HashMap<String, List<String>>(headers)
-                        .entrySet().stream()
-                        .filter(map -> !map.getKey().equalsIgnoreCase(AUTHORIZATION_HEADER))
-                        .collect(Collectors.toMap(p -> p.getKey(), p -> p.getValue()));
-                auditInfo.put(REST_REQUEST_HEADERS, headersClone);
-            } else {
-                auditInfo.put(REST_REQUEST_HEADERS, new HashMap<String, List<String>>(headers));
+            final Map<String, List<String>> headersClone = new HashMap<>(headers);
```

Comment:
-Is it necessary to create the clone even if `excludeSensitiveHeaders` is `false`?

Reply:
Input can be unmodifiable map. IMO its better to have consistency in what the method is setting for that attribute. It would assign unmodifiable map if excludeSensitiveHeaders is false or modifiable map if true. Retaining current behavior of setting modifiable map in either case.

Reply:
Why does it matter? Where and why it can be modified?

Reply:
As i ve mentioned in the PR body, retaining the current behavior as it was specifically doing that. 
This may help in building a redactor module.

Reply:
I don't see why is it necessary to keep the original behavior.  It should be redactor module responsibility to create a copy or filter original `Map` if necessary.